scipy/stats/_hypotests.py

```diff
@@ -1433,3 +1435,270 @@ def cramervonmises_2samp(x, y, method='auto'):
             p = max(0, 1. - _cdf_cvm_inf(tn))
 
     return CramerVonMisesResult(statistic=t, pvalue=p)
+
+
+class Tukey_HSDResult:
+    """Result of `scipy.stats.tukey_hsd`.
+
+    Attributes
+    ----------
+    statistic : float ndarray
+        The computed statistic of the test for each comparison. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+    pvalue : float ndarray
+        The associated p-value from the studentized range distribution. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+
+    Methods
+    -------
+    confidence_interval :
+        Compute the confidence interval for the significance level
+        specified in the test.
+
+
+    References
+    ----------
+    .. [1] NIST/SEMATECH e-Handbook of Statistical Methods, "7.4.7.1. Tukey's
+           Method."
+           https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
+           28 November 2020.
+    """
+
+    def __init__(self, statistic, pvalue, _nobs, _ntreatments, _stand_err):
+        self.statistic = statistic
+        self.pvalue = pvalue
+        self._ntreatments = _ntreatments
+        self._nobs = _nobs
+        self._stand_err = _stand_err
+
+    def confidence_interval(self, sig_level=.05):
+        """
+        Compute the confidence interval for the specified significance level.
+
+        Parameters
+        ----------
+        confidence_interval : float, optional
```

Comment:
```suggestion
        confidence_level: float, optional
```
And please change the default to 0.95 as in [`binomtest`](http://scipy.github.io/devdocs/reference/generated/scipy.stats.binomtest.html#scipy.stats.binomtest). The confidence level is the complement of the significance level.